src/target/riscv/riscv.c

```diff
@@ -3096,25 +3088,76 @@ static int riscv_write_phys_memory(struct target *target, target_addr_t phys_add
 	return tt->write_memory(target, phys_address, size, count, buffer);
 }
 
-static int riscv_write_memory(struct target *target, target_addr_t address,
-		uint32_t size, uint32_t count, const uint8_t *buffer)
+static int riscv_rw_memory(struct target *target, target_addr_t address, uint32_t size,
+		uint32_t count, uint8_t *read_buffer, const uint8_t *write_buffer)
 {
+	if (read_buffer && write_buffer) {
+		LOG_TARGET_ERROR(target, "Can't handle both read and write buffer.");
+		return ERROR_FAIL;
+	}
+
+	bool is_write = write_buffer ? true : false;
```

Comment:
Nitpick (it helps code readers a little):

```suggestion
	const bool is_write = write_buffer ? true : false;
```

Reply:
Thanks. Addressed.